### PR TITLE
Google Maps Heatmaps for iOS

### PR DESCRIPTION
**Pull request by david-cako**
### Does any other open PR do the same thing?

No.

### What issue is this PR fixing?

PR implements Google Maps's native heatmaps support on iOS (https://github.com/googlemaps/google-maps-ios-utils)

### How did you test this PR?

Tested on iOS Simulator (PR only targets Google Maps) by adding points to a heatmap.

### CODE REVIEW

**Comment by rborn** [comment]
@david-cako any chance to add it for android too to have a consistency between platforms?
also please add the docs needed for it   🤗

@alvelig 🐽

**Comment by david-cako** [comment]
@rborn For sure.  Give me some time to spool on it.

**Comment by christopherdro** [comment]
Related https://github.com/react-community/react-native-maps/pull/1549

**Comment by KentonParton** [comment]
@david-cako thank you for your implementation so far. I just wanted to find out if this has been added to Android and if this will be merged to the main branch sometime in the near future? Thanks!

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,7 @@
+.project
```
Comment (rborn) [inline .gitignore]: are we sure we need this file commited?

File: README.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+# david-cako/react-native-maps
```
Comment (rborn) [inline README.md]: Please change the readme  only with the functionality (lower in the examples or where their place is)

File: package.json [hunk 3]
```diff
@@ -1,61 +1,61 @@
 {
-  "name": "react-native-maps",
```
Comment (rborn) [inline package.json]: Why is package.json changed?  if it's format only please remove it from PR, if has anything new please make only that diff to be changed, not the whole file

File: lib/ios/AirGoogleMaps/AIRGoogleMapMarker.h [hunk 4]
```diff
@@ -31,7 +31,7 @@
 @property (nonatomic, copy) NSString *subtitle;
 @property (nonatomic, strong) UIColor *pinColor;
 @property (nonatomic, assign) CGPoint anchor;
-@property (nonatomic, assign) CGPoint calloutAnchor;
+@property(nonatomic, assign) CGPoint calloutAnchor;
```
Comment (rborn) [inline lib/ios/AirGoogleMaps/AIRGoogleMapMarker.h]: let's put back the space here :) to have the code consistent

**Comment by rborn** [comment]
@david-cako thank you  very much for this ❤️ LGTM but I need more eyes 
@alvelig @christopherdro 🐽


@david-cako  I have few more comments:
- can you please update/create an example on how to use the new heatmap?
- the readme file seems to have gradient polygon for ios, maybe update it with your addition?
- is calloutAnchor for both ios and android? and it's documented?

Thanks again 🤗

**Comment by david-cako** [comment]
Still working on android implementation, but will keep you posted.  Note that both the android and iOS implementation requires an additional native package.

Unfortunately, nothing is documented at this point as I have been actively developing on my fork.

**Comment by kostiantyn-solianyk** [comment]
Heatmap is a great implementation, looking forward to update. How long does it take?

**Comment by jakoblorz** [comment]
What is the state of this PR?

**Comment by david-cako** [comment]
@jakoblorz 

Gradient polylines, heatmaps, and onMapLoaded (tiles fully loaded) callback fully implemented for Google Maps on both iOS and Android on my fork.

**Comment by rborn** [comment]
@david-cako will you update this PR (when it's done) ?❤️

**Comment by abegehr** [comment]
@david-cako Awesome! 👍 Do you think your fork would be up for merging to master – who would review and decide this?
– This feature is highly needed!

**Comment by kostiantyn-solianyk** [comment]
any news?

**Comment by YeshanJay** [comment]
Any update on this guys? Would be great to have this feature. @david-cako  @rborn

**Comment by christopherdro** [comment]
Closing for https://github.com/react-native-community/react-native-maps/pull/2960

**Pull request closed without merging**